### DISCUSSION TITLE
Config/Memories Backup

### DISCUSSION THREAD

**Discussion by gdyuldin** [gdyuldin/x6100_gui#20] [Ideas]
It might be implemented by copying files to the internal storage during shutdown and restoring during power on.
Advantages:
* Config will survive during flashing new version of software on the SD card.
* Config will survive, if SD card will be died.
Disadvantages:
* Time of starting/turning off might be increased
* Internal storage has not so much free space (6.3 Gb)
* Backup might be deleted during flashing new Xiegu FW

I guess, copying only params.db, ADI log, and recorded messages might be enough. Recordings, screenshoots might not be copied.

**Comment by PCBenetton** [comment]
Maybe it could be a manual procedure.
Backup manually to the data partition of the SD card, where one can extract and save it somewhere safe.
After SD failure or after flashing new fw simply paste it to the data partition and then manually restore settings/memories.

And maybe do it all via the webserver.

**Comment by gdyuldin** [comment]
Config/Memories already stored to the DATA partition (FAT32) - so you can save it.
New image have no DATA partition, but it creates during first boot. So - you can copy all from DATA, flash new image, boot transceiver, turn it off, put SD card to computer and copy your backup back to DATA.

**Reply by PCBenetton** [reply]
I had no ideia. Never got the SD card back on the PC unless it was for flashing new fw. Thanks! Will back it up now!

**Comment by ghost** [comment]


**Reply by gdyuldin** [reply]
Alternative firmware is completely independent from OEM firmware, and using only SD card storage. shm is a temporary file storage filesystem, that uses RAM for the backing store.

**Comment by gdyuldin** [comment]
Another option for backup (proposed by stan kiwi on groups.io)

> Well, must repeat wifi config after each next SD firmware upgrade. Georgy, maybe add Backup button to webserver page. It would do sth like:
> tar -czf /mnt/x6100_r1cbu_private.tgz /etc/NetworkManager/system-connections/ /root/.ssh/
> After SD upgrage I will copy created .tgz to DATA part. and next boot will merge my private settings into new system.

I think it should also include params.db, recordings, etc. Application should be turned off before backup and turned on after backup (for database consistency).

**Comment by fventuri** [comment]
Georgy,
I took a quick look at Xiegu's script '/usr/share/emmc_sources/install_emmc.sh' last night, and it looks like the only three directories that Xiegu backs up during an upgrade of their firmware are '/usr/app_qt/voice', '/etc/xgradio' and '/etc/NetworkManager'.

Since I wasn't sure about the first two (i.e. Xiegu might change their layout or decide in a future firmware to rebuild them from scratch), but I am pretty sure the location and layout of  '/etc/NetworkManager' wouldn't change, I decided to use that to create a subdirectory called 'R1CBU', and store all the files from the data partition in there.

I actually mounted it under '/mnt' with a bind mount, so no other changes should be needed.

73,
Franco

**Comment by Tank-NY** [comment]
Backup should also include WiFi passwords which to my understanding are not in the Data partition and may not be in easy reach of a standard Windows machine, especially since Wifi is working for time sync, re-entering Wifi passwords each time there is an update is not efficient.
